packages/imperative/src/events/src/EventUtils.ts

```diff
@@ -205,14 +205,18 @@ export class EventUtils {
             // Accommodates for the delay between actual file change event and fsWatcher's perception
             //(Node.JS triggers this notification event 3 times)
             event.eventTime = EventUtils.getEventContents(event.eventFilePath).eventTime;
-            if (eeInst.eventTimes.get(eventName) !== event.eventTime && process.pid !== event.appProcId) {
+            let shouldProcessEvent = eeInst.eventTimes.get(eventName) !== event.eventTime;
+            eeInst.eventTimes.set(eventName, event.eventTime);
+            // Checks that event was not triggered by the same process
+            shouldProcessEvent &&= process.pid !== event.appProcId;
```

Comment:
Could lines 208 and 211 be combined? Combining the lines will also allow you to use `const` instead of `let`: 

```typescript
const shouldProcessEvent = eeInst.eventTimes.get(eventName) !== event.eventTime && process.pid !== event.appProcId;`
```